shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2488,7 +2497,18 @@ - (void)setSelectionRects:(NSArray*)encodedRects {
                                                          ? NSWritingDirectionLeftToRight
                                                          : NSWritingDirectionRightToLeft]];
   }
-  _activeView.selectionRects = rectsAsRect;
+
+  if (@available(iOS 17, *)) {
+    // Force UIKit to query the selectionRects again on iOS 17+
+    // This is to fix a bug on iOS 17+ where UIKit queries the outdated selectionRects after
+    // entering a character, resulting in auto-correction highlight region missing the last
+    // character.
+    [_activeView.inputDelegate textWillChange:_activeView];
+    _activeView.selectionRects = rectsAsRect;
+    [_activeView.inputDelegate textDidChange:_activeView];
+  } else {
+    _activeView.selectionRects = rectsAsRect;
```

Comment:
Rather than duplicate this line, which could cause drift over time, we should refactor to look more like the other method where there's a bool for sending updates.:

```
BOOL triggerTextUpdates = NO;
if (@available(iOS 17, *)) {
  // move explanatory comment here.
  triggerTextUpdates = YES;
}
if (triggerTextUpdates) {
  [_activeView.inputDelegate textWillChange:_activeView];
}
_activeView.selectionRects = rectsAsRect;
if (triggerTextUpdates) {
  [_activeView.inputDelegate textDidChange:_activeView];
}
```